docs/source/tutorials/memory_optimizations.rst

```diff
@@ -15,14 +15,14 @@ To make things easy, we've summarized these components in the following table:
    :widths: auto
 
    ":ref:`glossary_precision`", "You'll usually want to leave this as its default ``bfloat16``. If you're struggling with training stability or accuracy due to precision, fp32 may help, but will significantly increase memory usage and decrease training speed."
-   ":ref:`glossary_act_ckpt`", "Use when you're memory constrained and need to handle larger batch sizes or longer context lengths. Be aware that it may slow down training speed."
-   ":ref:`glossary_act_off`", "Similar to activation checkpointing, this can be used when memory constrained, but comes at the cost of training speed due to the overhead of moving tensors between GPU VRAM and CPU. This can also be used alongside activation checkpointing."
-   ":ref:`glossary_grad_accm`", "Helpful when memory-constrained to simulate larger batch sizes. Often preferable to activation checkpointing for better training speed."
-   ":ref:`glossary_low_precision_opt`", "When you need to further reduce memory usage beyond using ``bf16`` by reducing the precision in the optimizer states. Note that lower precision optimizers may reduce training stability/accuracy."
+   ":ref:`glossary_act_ckpt`", "Use when you're memory constrained and want to use a larger model, batch size or context lengths. Be aware that it will slow down training speed."
+   ":ref:`glossary_act_off`", "Similar to activation checkpointing, this can be used when memory constrained, but may decrease training speed due to the overhead of moving tensors between GPU VRAM and CPU. We minimize it by using a different stream, so you may not experience any slow down. This **should** be used alongside activation checkpointing."
+   ":ref:`glossary_grad_accm`", "Helpful when memory-constrained to simulate larger batch sizes. However, it is not compatible with optimizer in backward. Use it when training adapters, e.g. LoRA, which don't benefit from optimizer in backward, or when you can already fit a batch with you memory, but not enough of them.
+   ":ref:`glossary_low_precision_opt`", "When you have a large model (optimizer state is 2x model size) and need to further reduce memory. Note that lower precision optimizers may reduce training stability/accuracy."
```

Comment:
I feel like we are making these so-called summaries longer rather than shorter, which I do not like. Like honestly I have half a mind to just give columns like "memory", "tokens per second", "when to use" (maybe "constraints" too?) with up/down arrows for the first two columns and < 10 words on "when to use" (e.g. long sequence length, increase effective batch size, etc). The individual items further down should be responsible for getting more into the weeds.

Reply:
Also the offloading blurb is too handwavy. (Again would benefit from being moved down to the more detailed section to give it the proper treatment, as a first-time reader just trying to understand my options I definitely do not care that it's using a separate stream)

Reply:
> Like honestly I have half a mind to just give columns like "memory", "tokens per second", "when to use" (maybe "constraints" too?) with up/down arrows for the first two columns and < 10 words on "when to use" (e.g. long sequence length, increase effective batch size, etc).

I'd be up for this, happy to do it in a follow up.

Reply:
@SalmanMohammadi , i am gathering data to write a blog and add a table. After i have a draft, i would love to have your input. I should have it prob in a week.

Reply:
i simplified and removed info